src/app/component/ProductCard/ProductCard.component.js

```diff
@@ -90,13 +106,16 @@ export class ProductCard extends PureComponent {
 
     renderConfigurablePriceBadge() {
         const {
-            product: { type_id }
+            product: { type_id },
+            setSiblingsHavePriceBadge
         } = this.props;
 
         if (type_id !== CONFIGURABLE) {
             return null;
         }
 
+        setSiblingsHavePriceBadge();
```

Comment:
I believe this will cause extra reconciliation cycle triggers, although the rerenders will not be triggered due to using PureComponent. Checking whether the `setState` should be called either here or inside of that function itself will improve its impact on performance a tiny bit :)

Reply:
Added check in order to prevent multiple setState function calls.